[mod] simple theme: collaps/expand elements in the sidebar

return42 opened the pull request:
## What does this PR do?

Make elements in the sidebar collapse able.  Except infoboxes all elements in the sidebar are collapsed by default.

## Why is this change important?

By folding out the sidebar elements, the UI looks less cluttered.  Especially on small devices like smartphones, where the sidebar is above the results list, the UX should be improved [#2140].

## Related issues

Closes https://github.com/searxng/searxng/issues/2140

------

## before

![grafik](https://user-images.githubusercontent.com/554536/234080688-630c999d-8772-4df1-860a-6c48e3ecbeca.png)

## after

![grafik](https://user-images.githubusercontent.com/554536/234080825-4ccdc27c-ba81-4025-bc74-46ab3e919ce9.png)

return42 committed:
[mod] simple theme: collaps/expand elements in the sidebar

Make elements in the sidebar collapse able.  Except infoboxes all elements in
the sidebar are collapsed by default.

By folding out the sidebar elements, the UI looks less cluttered.  Especially on
small devices like smartphones, where the sidebar is above the results list, the
UX should be improved [1].

[1] https://github.com/searxng/searxng/issues/2140

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

mrpaulblack committed:
[mod] simple theme: build design for details (collapsables)

* set border top and bottom on sidebar collasables
* inrease peading on summary so its easier to click on mobile
* remove margins and add flex wrapper to normalize elements in sidebar

mrpaulblack committed:
[build] /static

mrpaulblack reviewed (approved):
Sorry for the long wait I pushed my review on top; If ok please merge :+1: 

main changes:
* use pointer cursor on details summary
* add border bottom and top on details elements
* increase padding on details summary to make it easier to click on mobile
* normalize the sidebar elements so they fit correctly and have the same margins

Everything LGTM, thanks :+1:

return42 committed:
[fix] simple theme: move engine alerts in case of no results into sidebar

If there were no results but errors in the engines then the error dialogs of the
engines was displayed in the result list.

With the new design errors of the engines should only be displayed in the
sidebar and at the same time duplications of the (template) code will be
avoided.

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>
